### PR TITLE
add definition for non-execution policy std::find

### PR DESCRIPTION
**Pull request by gress2**

### CODE REVIEW

File: include/find.hpp [hunk 1]
```diff
@@ -18,14 +18,52 @@
 
 namespace bit {
 
-// TODO
 template <class InputIt, class T>
 constexpr bit_iterator<InputIt> find(bit_iterator<InputIt> first,
```
Comment (bkille) [inline include/find.hpp:22]: So I don't think this will work with InputIt either. I think, though, that it can easily be modified to. Instead of iterating over a bit_iterator until `cursor != last`, you could create a word iterator `auto it = first.base()` and then iterate over it until `it == std::next(last.base(), !is_last_aligned)`, at which point you can use the same logic you have already been using.

From an optimization standpoint, I think the above works better anyways. It avoids having to call `std::min`, `get_word` and a few other arithmetic operations on every iteration. As another side note, I think `get_word` should only be used when we potentially need more than 2 words. If we only need 2 words, it would be faster to just use `shld`/`shrd` I think, since its the same logic just without extra stuff.

File: src/tests/find.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    bit_iterator<vec_iter_t> first2(vec2.begin());
+    bit_iterator<vec_iter_t> last2(vec2.end());
+
+    pos = bit::find(first2, last2, bit::bit0);
+    pos_expected = std::find(first2, last2, bit::bit0);
+
+    REQUIRE(pos == pos_expected);
+}
+
+TEMPLATE_TEST_CASE("find: is ok for virtual/misaligned words", "[find]",
```
Comment (bkille) [inline src/tests/find.hpp:63]: I think that until Vincent finishes his test framework, we should default to the `TEMPLATE_PRODUCT_TEST_CASE` that you introduced a few weeks ago. I would prefer it if we had somewhat more rigorous tests in place before his framework rolls out, but that's up to everyone. At the very least, we should be testing it on all possible iterator types.

File: include/is_sorted.hpp [hunk 3]
```diff
@@ -22,7 +22,38 @@ namespace bit {
...
     bit_iterator<ForwardIt> last) {
-    (first, last); 
+
+    using word_type = typename bit_iterator<ForwardIt>::word_type;
+    std::size_t num_digits = binary_digits<word_type>::value;
+    
+    bit_iterator<ForwardIt> cursor = first;
+    bool has_seen_1 = false;
+
+    while (cursor != last) {
```
Comment (bkille) [inline include/is_sorted.hpp:32]: Same comment as before. While here we have Fwd iterators, it is still faster to work with word iterators. In general, I think `get_word` should only be used when we containers of different sizes. Otherwise, we can avoid the overhead by calling `_shld` or, even better, not do that at all. In this case and in `find`, we can can treat the first and last word as edge cases, and the words inbetween we can then just deal with directly, instead of having to shift for every word.

File: src/tests/find.hpp (hunk 2 above)
Comment (gress2) [reply src/tests/find.hpp:63]: My thinking was to implement basic logic checks for now and revisit things after the testing tool is done. I guess it depends on the level of quality we want to enforce on master in the short term.

**Review by bkille — APPROVED** [review]
I will create LaTeX file.
